src/sentry/replays/usecases/ingest/event_parser.py

```diff
@@ -402,32 +514,44 @@ def as_trace_item_context(event_type: EventType, event: dict[str, Any]) -> Trace
             if response_size:
                 resource_attributes["response_size"] = response_size
 
-            return {
-                "attributes": resource_attributes,
-                "event_hash": uuid.uuid4().bytes,
-                "timestamp": float(payload["startTimestamp"]),
-            }
+            timestamp = _get_timestamp(payload, key="startTimestamp")
+            return _make_trace_item_context(resource_attributes, timestamp, event_type, event)
+
         case EventType.RESOURCE_SCRIPT | EventType.RESOURCE_IMAGE:
-            return {
-                "attributes": {
-                    "category": (
-                        "resource.script"
-                        if event_type == EventType.RESOURCE_SCRIPT
-                        else "resource.img"
-                    ),
-                    "size": int(event["data"]["payload"]["data"]["size"]),
-                    "statusCode": int(event["data"]["payload"]["data"]["statusCode"]),
-                    "decodedBodySize": int(event["data"]["payload"]["data"]["decodedBodySize"]),
-                    "encodedBodySize": int(event["data"]["payload"]["data"]["encodedBodySize"]),
-                    "url": as_string_strict(event["data"]["payload"]["description"]),
-                    "duration": float(event["data"]["payload"]["endTimestamp"])
-                    - float(event["data"]["payload"]["startTimestamp"]),
-                },
-                "event_hash": uuid.uuid4().bytes,
-                "timestamp": float(event["data"]["payload"]["startTimestamp"]),
+            payload = _get_in(event, ["data", "payload"])
+            payload_data = _get_in(payload, ["data"])
+
+            resource_attributes = {
+                "category": (
+                    "resource.script" if event_type == EventType.RESOURCE_SCRIPT else "resource.img"
+                ),
             }
+
+            _map_attrs(
+                resource_attributes,
+                payload_data,
+                size=("size", int),
+                statusCode=("statusCode", int),
+                decodedBodySize=("decodedBodySize", int),
+                encodedBodySize=("encodedBodySize", int),
+            )
+            _map_attr(resource_attributes, "url", payload.get("description"), as_string_strict)
+            if "endTimestamp" in payload and "startTimestamp" in payload:
+                _map_attr(
+                    resource_attributes,
+                    "duration",
+                    float(payload["endTimestamp"]) - float(payload["startTimestamp"]),
+                )
+
+            if not check_attr_dict_length(resource_attributes, event_type, event):
+                return None
+
+            timestamp = _get_timestamp(payload, key="startTimestamp")
```

Comment:
Its okay to raise if the key doesn't exist. It will be caught and logged. There's no recovery possible after safe extraction.